custom/git-app-deployer/main.go

```diff
@@ -0,0 +1,169 @@
+package main
+
+import (
+	"bytes"
+	"flag"
+	"os/exec"
+	"time"
+
+	"github.com/litmuschaos/test-tools/pkg/log"
+	"github.com/openebs/maya/pkg/util/retry"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func main() {
+
+	namespace := "sock-shop"
+
+	filePath, timeout := GetData()
+
+	config, err := getKubeConfig()
+	if err != nil {
+		panic(err.Error())
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	log.Info("[Status]: App-deployer is started...")
+	log.Infof("[Status]: App-deployer filePath is %v", filePath)
+
+	if err := CreateNamespace(clientset, namespace); err != nil {
+		log.Errorf("Failed to create the namespace, err: %v", err)
+		return
+	}
+	log.Infof("[Status]: %v namespace is successfully created!", namespace)
+
+	if err := CreateSockShop("/var/run/"+filePath, namespace); err != nil {
+		log.Errorf("Failed to install sock-shop, err: %v", err)
+		return
+	}
+	log.Info("[Status]: Sock Shop applications is successfully created!")
+
+	if err := CheckApplicationStatus(namespace, "app=sock-shop", timeout, 2, clientset); err != nil {
+		log.Errorf("err: %v", err)
+		return
+	}
+
+}
+
+// GetKubeConfig function derive the kubeconfig
+func getKubeConfig() (*rest.Config, error) {
+	kubeconfig := flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
+	flag.Parse()
+	// It uses in-cluster config if kubeconfig path is not specified
+	config, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
+	return config, err
+}
+
+//GetData derive the sock-shop filePath and timeout
+//it derive the filePath based on sock-shop scenario(week vs resilient)
+func GetData() (string, int) {
+	typeName := flag.String("typeName", "", "absolute type of the sock-shop")
+	flag.Parse()
+
+	if *typeName == "" || *typeName == "weak" {
+		return "weak-sock-shop.yaml", 200
+	}
+	return *typeName + "-sock-shop.yaml", 200
```

Comment:
This hardcoded timeout can be a problem. can you fetch this from an env and set some default value(i.e 300s)

Reply:
Yes updated it to the ENV variable